crates/nargo_cli/src/manifest.rs

```diff
@@ -1,13 +1,234 @@
+/// supply a source for the dependency
+pub(super) enum Dependency {
+    Github { git: String, tag: String },
+    Path { path: String },
+}
+
+impl PackageConfig {
+    fn to_default_package(&self, root_path: &Path) -> Result<Package, ManifestError> {
+        let name = match self.package.name.as_ref() {
+            Some(name) => {
+                Some(CrateName::new(name).map_err(|_| ManifestError::InvalidPackageName)?)
+            }
+            None => None,
+        };
+
+        let mut dependencies: BTreeMap<CrateName, package::Dependency> = BTreeMap::new();
+        for (name, dep) in self.dependencies.iter() {
+            let name = CrateName::new(name).map_err(|_| ManifestError::InvalidPackageName)?;
+            let resolved_dep = resolve_dependency(root_path, dep)?;
+
+            dependencies.insert(name, resolved_dep);
+        }
+
+        Ok(Package { root_path: root_path.to_path_buf(), is_default: true, name, dependencies })
+    }
+    fn to_package(&self, root_path: &Path) -> Result<Package, ManifestError> {
+        let name = match self.package.name.as_ref() {
+            Some(name) => {
+                Some(CrateName::new(name).map_err(|_| ManifestError::InvalidPackageName)?)
+            }
+            None => None,
+        };
+
+        let mut dependencies: BTreeMap<CrateName, package::Dependency> = BTreeMap::new();
+        for (name, dep) in self.dependencies.iter() {
+            let name = CrateName::new(name).map_err(|_| ManifestError::InvalidPackageName)?;
+            let resolved_dep = resolve_dependency(root_path, dep)?;
+
+            dependencies.insert(name, resolved_dep);
+        }
+
+        Ok(Package { root_path: root_path.to_path_buf(), is_default: false, name, dependencies })
+    }
+}
+
+fn from_file(root_path: &Path) -> Result<NargoToml, ManifestError> {
     let toml_as_string =
-        std::fs::read_to_string(&path_to_toml).expect("ice: path given for toml file is invalid");
+        std::fs::read_to_string(root_path).expect("ice: path given for toml file is invalid");
+    let manifest = toml_as_string.try_into()?;
+    Ok(manifest)
+}
+
+pub(super) fn resolve_workspace(toml_path: &Path) -> Result<Workspace, ManifestError> {
+    let root_path = toml_path.parent().expect("TODO");
+    let manifest = from_file(toml_path)?;
+
+    let workspace = match manifest {
+        NargoToml::Package(pkg_config) => {
+            let member = pkg_config.to_default_package(root_path)?;
+            Workspace { members: vec![member] }
+        }
+        NargoToml::Workspace { workspace } => {
+            let mut saw_default = false;
+            let mut members = Vec::new();
+            for member_path in workspace.members {
+                let root_path = root_path.join(&member_path);
+                // TODO: All these calls should just work on a root_path (directory) and
+                // then we can `find_package_manifest` in `from_file` (needs a better name)
```

Comment:
We'll need to have a variant of this function which doesn't search through parents however.

Reply:
Done